create-presigned-download-url/src/main/java/no/unit/nva/download/publication/file/CreatePresignedDownloadUrlHandler.java

```diff
@@ -81,43 +76,77 @@ protected Integer getSuccessStatusCode(Void input, CreatePresignedDownloadUrlRes
         return HttpStatus.SC_OK;
     }
 
-    /**
-     * Get valid file from publication.
-     *
-     * @param fileIdentifier fileIdentifier
-     * @param fileSet fileSet
-     * @return valid file
-     * @throws ApiGatewayException exception thrown if valid file not present
-     */
-    private File getValidFile(UUID fileIdentifier, FileSet fileSet) throws ApiGatewayException {
+    private Publication fetchPublicationForAuthorizedUser(RequestInfo requestInfo) throws ApiGatewayException {
+        Publication publication = fetchPublication(requestInfo);
+        authorize(requestInfo, publication);
+        return publication;
+    }
 
-        Optional<List<File>> files = Optional.ofNullable(fileSet.getFiles());
+    private File fetchFile(RequestInfo requestInfo, Publication publication) throws ApiGatewayException {
```

Comment:
For a future tech debt task: Move this into business object for Publication to avoid data jealousy.

Reply:
We can create a task (Spike) on how put the business logic in the the objects.